index.js

```diff
@@ -303,9 +328,23 @@ const parseArgs = (config = kEmptyObject) => {
         }
       }
 
-      if (ObjectHasOwn(optionConfig, 'multiple')) {
+      const hasMultipleFlag = ObjectHasOwn(optionConfig, 'multiple');
+      if (hasMultipleFlag) {
         validateBoolean(optionConfig.multiple, `options.${longOption}.multiple`);
       }
+
+      if (ObjectHasOwn(optionConfig, 'defaultValue')) {
+        const defaultValue = objectGetOwn(optionConfig, 'defaultValue');
+        if (optionType === 'string' && !hasMultipleFlag) {
+          validateString(defaultValue, `options.${longOption}.defaultValue`);
+        } else if (optionType === 'string' && hasMultipleFlag) {
+          validateStringArray(defaultValue, `options.${longOption}.defaultValue`);
+        } else if (optionType === 'boolean' && !hasMultipleFlag) {
+          validateBoolean(defaultValue, `options.${longOption}.defaultValue`);
+        } else if (optionType === 'boolean' && hasMultipleFlag) {
+          validateBooleanArray(defaultValue, `options.${longOption}.defaultValue`);
+        }
```

Comment:
Regarding the `hasMultipleFlag` variable: I may be reading this wrong, but shouldn't we be checking these conditions against the value of the `multiple` option rather than its presence in the config?

```js
if (ObjectHasOwn(optionConfig, 'multiple')) {
  validateBoolean(optionConfig.multiple, `options.${longOption}.multiple`);
}

if (ObjectHasOwn(optionConfig, 'default')) {
  const defaultOption = objectGetOwn(optionConfig, 'default');
  const multipleOption = objectGetOwn(optionConfig, 'multiple');

  if (optionType === 'string' && !multipleOption) {
    validateString(defaultOption, `options.${longOption}.default`);

  } else if (optionType === 'string' && multipleOption) {
    validateStringArray(defaultOption, `options.${longOption}.default`);

  } else if (optionType === 'boolean' && !optionMultiple) {
    validateBoolean(defaultOption, `options.${longOption}.default`);

  } else if (optionType === 'boolean' && multipleOption) {
    validateBooleanArray(defaultOption, `options.${longOption}.default`);
  }
}
```

Reply:
Nice catch! Fixed and I added a test for this case!

Reply:
Awesome!

I did notice the update [removed the `hasOwn` check for `multiple`](https://github.com/pkgjs/parseargs/pull/142/commits/461365696be2fee4f053a7cc5d25fb4f8a1b7b38#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346L332-L334). I believe this should still be included as the `multiple` config is optional (see my above example). Note: The following `objectGetOwn(optionConfig, 'multiple')` will still function as expected (even if `'multiple'` is omitted) as the [`objectGetOwn` util also checks `hasOwn`](https://github.com/pkgjs/parseargs/blob/df925dd7ebe23cc7b2120b7a629ca6b8a97aa7ff/utils.js#L26-L27).

Reply:
Awesome!

I did notice the update introduces a similar issue when [validating the `multiple` option itself](https://github.com/pkgjs/parseargs/pull/142/commits/461365696be2fee4f053a7cc5d25fb4f8a1b7b38#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R331) e.g. (validating the presence of `multiple` instead of its value).

Since [`objectGetOwn` also checks `hasOwn`](https://github.com/pkgjs/parseargs/blob/df925dd7ebe23cc7b2120b7a629ca6b8a97aa7ff/utils.js#L26-L27), I think we can just pull up `const multipleOption = objectGetOwn(optionConfig, 'multiple')` as follows. Thoughts?
```js
const multipleOption = objectGetOwn(optionConfig, 'multiple');

if (ObjectHasOwn(optionConfig, 'multiple')) {
  validateBoolean(multipleOption, `options.${longOption}.multiple`);
}

if (ObjectHasOwn(optionConfig, 'default')) {
  const defaultOption = objectGetOwn(optionConfig, 'default');

  if (optionType === 'string' && !multipleOption) {
    validateString(defaultOption, `options.${longOption}.default`);
  ...
}
```

Reply:
Sorry 🤦🏽 I wanted to restore the previous code but I miss the `if` check.

Anyway with your suggestion we will run the `objectGetOwn(optionConfig, 'multiple');` once, so it is good to add it.

Working on it

Reply:
No worries! Thanks for addressing 👍